assimilation_code/modules/assimilation/assim_tools_mod.pf.f90

```diff
@@ -1604,7 +1910,90 @@ subroutine filter_assim(ens_handle, obs_ens_handle, obs_seq, keys,           &
       call error_handler(E_MSG,'',msgstring)
    endif
 
-endif
+endif ! KDDM
+
+! Perform RTPS step for hybrid
```

Comment:
this new RTPS section is for filter_kind 1, not the PF - is it yet a different experimental part of the code?

Reply:
the code is a hybrid, so for the last iteration it switches to an EAKF 
```
1987 filter_kind = 1
```

Reply:
but if filter_kind starts out as 1 it also goes into this code.  i did see the comment at the top of this file saying only run this version with filter_kind == 9 so i understand this isn't the integrated version.  for now it might be worth adding a line in the code after the comment at 1915 about the last iteration switching to EAKF.

Reply:
This part can definitely be cleaned up. I am currently using RTPS on the EAKF portion of the hybrid PF-EAKF update, but have it working on the EAKF alone for testing purposes. I advise users to not use this module for filter_kind other than 9 and with inflation turned off.

Reply:
added comment about hybrid using filter_kind=1 for last iteration.